realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +342,41 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread util transactions are committed by other threads to this Realm or
+     * {@link #stopWaitForChange()} gets called.
+     *
+     * @return {@code true} if transactions are committed to this Realm, {@code false} if
+     * {@link #stopWaitForChange()} gets called.
+     * @throws IllegalStateException if attempting to wait within a transaction.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        sharedGroupManager.setWaitForChangeEnabled(true);
+        // add this to wait list for batch stop
+        addRealmToWaitList(this);
+        boolean hasChanged = sharedGroupManager.waitForChange();
+        // remove this from wait list
+        removeRealmFromWaitList(this);
+        if (hasChanged) {
+            sharedGroupManager.advanceRead();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes {@link #waitForChange()} return {@code false} immediately. This method is supposed to
+     * be called by another thread different from the one created the Realm.
+     */
+    public void stopWaitForChange() {
+        // remove this from wait list
+        removeRealmFromWaitList(this);
```

Comment:
Is this necessary? Anyway it will be removed by line 362. same for the stop all.